keras/engine/training.py

```diff
@@ -2503,3 +2503,8 @@ def predict_generator(self, generator, steps=None,
             return [out[0] for out in all_outs]
         else:
             return [np.concatenate(out) for out in all_outs]
+
+# We need to overload the Keras Model class to handle MXNet model building activities.
+# get_model() returns the reference to MXNetModel that inherits and extends keras.engine.Model.
+if K.backend() == 'mxnet':
+    Model = K.get_model()
```

Comment:
This breaks the general philosophy of `keras-team/keras`, which is to have a shared codebase across different backends (which is possible as long as each backend implements the Keras backend API). Why do you think this  is necessary?

We've managed to do it across Theano and TF without much issues, even they work quite differently (e.g. sessions, graphs, etc). Surely you can manage any custom functionality you need at the level of `K.Function()`.